File: Python/Kode Program/Pasien.py
from Manusia import Manusia
from Obat import Obat

class Pasien(Manusia):
    def __init__(self, NIK, nama, umur, jenisKelamin, golDar, diagnosa):
        super().__init__(NIK, nama, umur, jenisKelamin)
        self.golDar = golDar
        self.diagnosa = diagnosa
        self.listObat = []   

    # Getter & Setter
    def getGolDar(self):
        return self.golDar

    def setGolDar(self, golDar):
        self.golDar = golDar

    def getDiagnosa(self):
        return self.diagnosa

    def setDiagnosa(self, diagnosa):
        self.diagnosa = diagnosa

    # Tambah obat ke list
    def addObat(self, obat: Obat):
        self.listObat.append(obat)

    # Print info pasien
    def printPasien(self):
        print(f"NIK : {self.getNIK()}, "
              f"Nama : {self.getNama()}, "
              f"Umur : {self.getUmur()}, "
              f"Jenis Kelamin : {self.getJenisKelamin()}, "
              f"Golongan Darah : {self.getGolDar()}, "
              f"Diagnosa : {self.getDiagnosa()}")
        
        if not self.listObat:
            print("  Tidak ada obat.")
        else:
            print("  Daftar Obat :")
            for o in self.listObat:
                print("  - ", end="")
                o.printObat()
        
        print("")

File: CPP/Kode Program/Pasien.cpp
#include <iostream>
#include <vector>
#include "Manusia.cpp"
#include "Obat.cpp"

using namespace std;

//Class Pasien extends Class Manusia
class Pasien : public Manusia{
    private:
        //Atribut Class Pasien
        string golDar;
        string diagnosa;
        vector<Obat> listObat;

    public:
        //Constructor
        Pasien(int NIK, string nama, string umur, string jenisKelamin, string golDar, string diagnosa) : Manusia(NIK, nama, umur, jenisKelamin){
            this->golDar = golDar;
            this->diagnosa = diagnosa;
        }

        //Getter dan setter
        string getGolDar(){
            return this->golDar;
        }

        void setGolDar(string golDar){
            this->golDar = golDar;
        }

        string getDiagnosa(){
            return this->diagnosa;
        }

        void setDiagnosa(string diagnosa){
            this->diagnosa = diagnosa;
        }

        //Method untuk menambahkan obat ke list obat, jadi resep obat seorang pasien
        void addObat(const Obat& o){
            listObat.push_back(o);
        }

        //Cetak info / data dari seorang pasien beserta obatnya
        void printPasien(){
            cout << "NIK : " << getNIK()
            << ", Nama : " << getNama()
            << ", Umur : " << getUmur()
            << ", Jenis Kelamin : " << getJenisKelamin()
            << ", Golongan Darah : " << getGolDar() 
            << ", Diagnosa : " << getDiagnosa() << endl;

            //Jika tidak ada obat / tidak diresepkan obat
            if(listObat.empty()){
                cout << "  Tidak ada obat." << endl;
            //Jika ada obat
            }else{
                cout << "  Daftar Obat :" << endl;
                for(auto& o : listObat){
                    cout << "  - ";
                    o.printObat();
                }
            }

            cout << "" << endl;
        }

        //Destructor
        ~Pasien(){
        }
};